js/modules/k6/grpc/client.go

```diff
@@ -212,15 +205,46 @@ func (t transportCreds) ClientHandshake(ctx context.Context,
 	return out, auth, err
 }
 
+func (c *Client) dial(
+	ctx context.Context,
+	addr string,
+	reflect bool,
+	options ...grpc.DialOption,
+) error {
+	opts := []grpc.DialOption{
+		grpc.WithBlock(),
+		grpc.FailOnNonTempDialError(true),
+		grpc.WithStatsHandler(c),
+		grpc.WithReturnConnectionError(),
+	}
+	opts = append(opts, options...)
+
+	var err error
+	c.conn, err = grpc.DialContext(ctx, addr, opts...)
+	if err != nil {
+		return err
+	}
+
+	if !reflect {
+		return nil
+	}
+
+	err = c.reflect(ctx)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
 // Connect is a block dial to the gRPC server at the given address (host:port)
-// nolint:funlen,cyclop
-func (c *Client) Connect(ctxPtr *context.Context, addr string, params map[string]interface{}) (bool, error) {
-	state := lib.GetState(*ctxPtr)
+func (c *Client) Connect(addr string, params map[string]interface{}) (bool, error) {
+	state := c.vu.State()
 	if state == nil {
 		return false, errConnectInInitContext
 	}
-	isPlaintext, reflect, timeout := false, false, 60*time.Second
 
+	isPlaintext, reflect, timeout := false, false, 60*time.Second
 	for k, v := range params {
```

Comment:
What if we move this to a `parseParams` method, it reduces the function length a bit and looks safe, wdyt?